### PR TITLE
PublicDashboards: Add RBAC to secured endpoints

### PR DESCRIPTION
**Pull request by evictorero**
**What this PR does / why we need it**:
Secure non public endpoints for public dashboards

**Which issue(s) this PR fixes**:

Fixes https://github.com/grafana/grafana-enterprise/issues/4459

**Special notes for your reviewer**:

We have to secure two endpoints for public dashboards:
* GET Public Dashboard config
**RBAC off**
Any user signed in as a viewer
**RBAC on**
Any user with `dashboards.ActionDashboardsRead`

* POST (save or update) Public Dashboard config
**RBAC off**
Only Org Admins
**RBAC on**
Any user with `dashboards.ActionDashboardPublicWrite` (new action and role)

### CODE REVIEW

File: pkg/services/publicdashboards/api/api.go [hunk 1]
```diff
@@ -51,18 +51,23 @@ func ProvideApi(
...
 
+	// public endpoints
 	api.RouteRegister.Get("/api/public/dashboards/:accessToken", routing.Wrap(api.GetPublicDashboard))
 	api.RouteRegister.Post("/api/public/dashboards/:accessToken/panels/:panelId/query", routing.Wrap(api.QueryPublicDashboard))
 
 	// Create/Update Public Dashboard
-	api.RouteRegister.Get("/api/dashboards/uid/:uid/public-config", auth(reqSignedIn, accesscontrol.EvalPermission(dashboards.ActionDashboardsWrite)), routing.Wrap(api.GetPublicDashboardConfig))
-	api.RouteRegister.Post("/api/dashboards/uid/:uid/public-config", auth(reqSignedIn, accesscontrol.EvalPermission(dashboards.ActionDashboardsWrite)), routing.Wrap(api.SavePublicDashboardConfig))
+	api.RouteRegister.Get("/api/dashboards/uid/:uid/public-config",
+		auth(middleware.ReqSignedIn, accesscontrol.EvalPermission(dashboards.ActionDashboardsRead)),
```
Comment (kalleep) [inline pkg/services/publicdashboards/api/api.go:65]: I guess the scope should be evaluated as well? Then a user could only read public config of a dashboard they have read access to. Or is this check done in another place then the middleware?

File: pkg/services/publicdashboards/api/api.go [hunk 2]
```diff
@@ -51,18 +51,23 @@ func ProvideApi(
...
+		routing.Wrap(api.GetPublicDashboardConfig))
+
+	api.RouteRegister.Post("/api/dashboards/uid/:uid/public-config",
+		auth(middleware.ReqOrgAdmin, accesscontrol.EvalPermission(dashboards.ActionDashboardPublicWrite)),
```
Comment (kalleep) [inline pkg/services/publicdashboards/api/api.go:69]: Same as abov

**Review by kalleep — COMMENTED** [review]
Looks good, left at few comments

File: pkg/services/publicdashboards/api/api.go (hunk 1 above)
Comment (evictorero) [reply pkg/services/publicdashboards/api/api.go]: You are right, added the scope. If a user has access to a dashboard, we want to allow it to see the public config.

File: pkg/services/publicdashboards/api/api.go (hunk 2 above)
Comment (evictorero) [reply pkg/services/publicdashboards/api/api.go]: Added scope

**Review by kalleep — APPROVED** [review]

**Pull request merged**